apps/web/app/api/webhooks/retell-ai/route.ts

```diff
@@ -0,0 +1,200 @@
+import { defaultResponderForAppDir } from "app/api/defaultResponderForAppDir";
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import { Retell } from "retell-sdk";
+import { z } from "zod";
+
+import { CreditService } from "@calcom/features/ee/billing/credit-service";
+import logger from "@calcom/lib/logger";
+import { safeStringify } from "@calcom/lib/safeStringify";
+import { prisma } from "@calcom/prisma";
+
+const log = logger.getSubLogger({ prefix: ["retell-ai-webhook"] });
+
+const RetellWebhookSchema = z.object({
+  event: z.enum(["call_started", "call_ended", "call_analyzed"]),
+  call: z
+    .object({
+      call_id: z.string(),
+      agent_id: z.string().optional(),
+      from_number: z.string(),
+      to_number: z.string(),
+      direction: z.enum(["inbound", "outbound"]),
+      call_status: z.string(),
+      start_timestamp: z.number(),
+      end_timestamp: z.number().optional(),
+      disconnection_reason: z.string().optional(),
+      metadata: z.record(z.any()).optional(),
+      retell_llm_dynamic_variables: z.record(z.any()).optional(),
+      transcript: z.string().optional(),
+      opt_out_sensitive_data_storage: z.boolean().optional(),
+      call_cost: z
+        .object({
+          product_costs: z
+            .array(
+              z.object({
+                product: z.string(),
+                unitPrice: z.number().optional(),
+                cost: z.number().optional(),
+              })
+            )
+            .optional(),
+          total_duration_seconds: z.number().optional(),
+          total_duration_unit_price: z.number().optional(),
+          total_one_time_price: z.number().optional(),
+          combined_cost: z.number().optional(),
+        })
+        .optional(),
+      call_analysis: z
+        .object({
+          call_summary: z.string().optional(),
+          in_voicemail: z.boolean().optional(),
+          user_sentiment: z.string().optional(),
+          call_successful: z.boolean().optional(),
+          custom_analysis_data: z.record(z.any()).optional(),
+        })
+        .optional(),
+    })
+    .passthrough(),
+});
+
+async function handleCallAnalyzed(callData: any) {
```

Comment:
This route is responsible for deducting credits after we get call_analyzed event from retell ai